src/main/java/org/datadog/jmxfetch/JMXComplexAttribute.java

```diff
@@ -34,7 +34,7 @@ private void populateSubAttributeList(Object attributeValue) {
             for (String key : data.getCompositeType().keySet()) {
                 this.subAttributeList.put(key, new HashMap<String, Object>());
             }
-        } else if ("java.util.HashMap".equals(attributeType)) {
+        } else if (("java.util.HashMap".equals(attributeType)) || ("java.util.Map".equals(attributeType))){
             HashMap<String, Double> data = (HashMap<String, Double>) attributeValue;
```

Comment:
(we could probably just cast `attributeValue` to a `Map` right?)

same applies to your change line 86